### ISSUE TITLE
Window capture on web browsers offsets after fullscreening browser and unfullscreening

### ISSUE THREAD

**Issue by Jared554** [obsproject/obs-studio#10406]
### Operating System Info

Windows 10

### OBS Studio Version

30.1.0

### OBS Studio Log URL

20:46:33.551: User added source 'Browser' (browser_source) to scene 'Scene' 20:46:36.569: User Removed source 'Browser' (browser_source) from scene 'Scene' 20:46:42.322: User added source 'Color Source' (color_source_v3) to scene 'Scene' 20:47:04.948: User added source 'Window Capture' (window_capture) to scene 'Scene' 20:47:05.010: [window-capture: 'Window Capture'] update settings: 20:47:05.010: 	executable: firefox.exe 20:47:05.010: 	method selected: Automatic 20:47:05.010: 	method chosen: WGC 20:47:05.010: 	force SDR: false

### Expected Behavior

Adding a window capture then selecting chrome or firefox, when you fullscreen chrome or firefox and then un-fullscreen chrome or firefox, it should go back to the exact same position and cropping as before.

### Current Behavior

When using chrome or firefox with window capture, if you fullscreen then un-fullscreen it will offset the top and left side of the source and cut off a bit

### Steps to Reproduce

1. add window capture
2. select chrome or firefox window
3. fullscreen chrome or firefox then un-fullscreen
4. the window will now be offset on obs for no reason

**Comment by Jared554** [comment]
https://github.com/obsproject/obs-studio/assets/94695228/1a5ae3db-2f12-4d05-a527-71b3955079fa

**Comment by Fenrirthviti** [comment]
Do you have your Windows Taskbar on the top or side of your display?

**Comment by Jared554** [comment]
> Do you have your Windows Taskbar on the top or side of your display?

the taskbar is on the bottom of the display

**Comment by Jared554** [comment]
i just checked and this doesnt happen on bitbit window capture firefox, this might be something with the yellow borders on the windows 10 capture, just a thought, but hopefully this can get fixed soon

**Comment by RytoEX** [comment]
> i just checked and this doesnt happen on bitbit window capture firefox, this might be something with the yellow borders on the windows 10 capture, just a thought, but hopefully this can get fixed soon

Unfortunately, Windows adds those borders. On Windows 11, you can turn them off. On Windows 10, you cannot turn them off. We'll try to investigate further, but this may be outside of our control if the borders being drawn is causing it.

**Comment by XMhat** [comment]
This actually happens on MacOS too, when capturing native full-screen windows on certain apps like Firefox, RetroArch, etc, with the Window Capture method, we're having to crop black borders as well. It's quite frustrating and it would be nice to have access to an auto cropping shader like the UAE emulator has.

**Comment by XMhat** [comment]
This actually happens on MacOS too, when capturing native full-screen windows on certain apps like Firefox, RetroArch, etc, with the Window Capture method, we're having to crop black borders from the top as well. It's quite frustrating and it would be nice to have access to an auto cropping shader like the UAE emulator has.

**Comment by RytoEX** [comment]
> This actually happens on MacOS too, when capturing native full-screen windows on certain apps like Firefox, RetroArch, etc, with the Window Capture method, we're having to crop black borders from the top as well. It's quite frustrating and it would be nice to have access to an auto cropping shader like the UAE emulator has.

The Windows and macOS capture source code is completely separate. It is unlikely, though admittedly possible, that your issue is the same, despite appearances.

**Comment by RytoEX** [comment]
Does this also occur in older versions of OBS Studio?

**Comment by Jared554** [comment]
> Does this also occur in older versions of OBS Studio?

yes this has happened for years, i reported an issue for this same offset issue happening in 2021 when the browser window was just minimized and maximized, but that got fixed, and i realized maybe a month ago it happens on fullscreening and unfullscreening, and my obs was also quite out of date at the time
